creator/ingest_runs/tasks/validation_run.py

```diff
@@ -1,82 +1,171 @@
 import os
+import jsonpickle
 import logging
-import django_rq
+from pprint import pformat
+import sys
+
+import pandas
 from django.conf import settings
 from django.core.files.base import ContentFile
 from django_s3_storage.storage import S3Storage
 
+from kf_lib_data_ingest.etl.extract.utils import Extractor
+from kf_lib_data_ingest.validation.data_validator import (
+    Validator as DataValidator,
+)
+from kf_lib_data_ingest.validation.reporting.markdown import (
+    MarkdownReportBuilder,
+)
+
 from creator.decorators import task
 from creator.ingest_runs.models import ValidationRun, ValidationResultset
+from creator.files.models import Version
+from creator.analyses.file_types import FILE_TYPES
+from creator.analyses.analyzer import extract_data
 
 S3_STORAGE = "django_s3_storage.storage.S3Storage"
 
 logger = logging.getLogger(__name__)
 
 
-def validate_file_versions(validation_run):
+def clean_and_map(version):
     """
-    TODO - Replace later. Currently counts to 10 then returns
+    Load tabular data from file version into a DataFrame and then use
+    ingest lib's extraction utility to clean the values and map the column
+    names to the standard set of concepts expected by the validator
+    """
+    logger.info(f"Performing extraction to clean and map file {version}")
 
-    Upload validation results and report files to storage backend
+    # Load ingest extract operations for this file
+    if not version.root_file:
+        raise Exception(
+            f"Validation failed. Version {version} missing root_file"
+        )
+    ec = version.extract_config_path
+    if not ec:
+        logger.warning(
+            f"⚠️  Could not find an extract config for file version {version} "
+            f"with file type: {version.root_file.file_type}. This file will "
+            "not be included in validation."
+        )
+        return None
```

Comment:
Why not raise here instead?

Reply:
It depends on what kind of behavior we want. Should we fail validation if one or more of the files in the review doesn't have an extract config? Or should we allow validation to continue for the files that do have extract configs?

Reply:
To me, the purpose of this function is to clean and map a data file. If an extract config cannot be found, then it cannot complete its purpose and fails. Similarly, if something goes wrong with extracting the data, it also fails. However, one case is encoded in an exception and the other is encoded by the `None` value. 
If you instead were to raise exceptions specific to the problem, the calling context (`validate_file_versions()`) would be able to tell why exactly it failed and act accordingly (continue validation if no extract config, fail out if extract fails). Furthermore, a developer reading the `validate_file_versions()` function would immediately understand exactly what case `except ExtractConfigDoesNotExist` is handling without having to read through `clean_and_map()` to understand why `None` is being returned.

Reply:
Oh I think I misunderstood your original comment. I agree with your feedback. Updated